packages/api/cms-api/src/common/filter/mikro-orm.spec.ts

```diff
@@ -1,38 +1,96 @@
 import { BooleanFilter } from "./boolean.filter";
 import { DateFilter } from "./date.filter";
-import { filtersToMikroOrmQuery, filterToMikroOrmQuery, searchToMikroOrmQuery } from "./mikro-orm";
+import { filtersToMikroOrmQuery, filterToMikroOrmQuery, searchToMikroOrmQuery, splitSearchString } from "./mikro-orm";
 import { NumberFilter } from "./number.filter";
 import { StringFilter } from "./string.filter";
 
+describe("splitSearchString", () => {
```

Comment:
With this you're actually testing an implementation detail. Wouldn't it be better to only test the public API (= searchToMikroOrmQuery)?

Reply:
I wouldn't view it that way. I'd say `splitSearchString` adds functionality that is non-trivial and can hardly be grasped from looking at the code alone. So IMO testing it is fine and it would be less.

Reply:
Testing the code is fine, but the test should be against the public API. Exporting a private function just to be able to test it seems like an anti-pattern to me. 

For instance, let's take this test:

```ts
it("should split a simple space-separated string", () => {
    const input = "This is a test";
    const expected = ["%This%", "%is%", "%a%", "%test%"];
    expect(splitSearchString(input)).toEqual(expected);
});
```

 IMO a better test would be:

```ts
it("should split a simple space-separated string", () => {
  expect(searchToMikroOrmQuery("This is a test", ["title"])).toStrictEqual({
    $and: [
      {
        $or: [
          {
            title: {
              $ilike: "TODO", // Insert result of splitSearchString here
            },
          },
        ],
      },
    ],
  });
});

```

Reply:
IMO testing it via integration tests doesn't help to understand it properly, since the expected outcome could be rather confusing for someone new. I think unit test are a better fit in this case.